src/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -63,5 +63,42 @@ public void Pop()
             Debug.Assert(_index > 0);
             Current = _previous[--_index];
         }
+
+        // Return a property path in the form of: [FullNameOfType].FirstProperty.SecondProperty.LastProperty
+        public string PropertyPath
+        {
+            get
+            {
+                StringBuilder path = new StringBuilder();
+
+                if (_previous == null || _index == 0)
+                {
+                    path.Append($"[{Current.JsonClassInfo.Type.FullName}]");
+                }
+                else
+                {
+                    path.Append($"[{_previous[0].JsonClassInfo.Type.FullName}]");
```

Comment:
Is it guaranteed for `_previous.Count > 0`, here (and `_index > 0`)?

Reply:
Yes. `_index` is actually the next free index.